### PR TITLE
Upgrade to Rocket 0.5 and switch to stable Rust

### PR DESCRIPTION
**Pull request by coreyja**
### TLDR Changes

- Upgrades to Rocket `0.5.0-rc.1`
- Switch Replit to use `replit.nix` to get a newer Rust compiler

## Fixing the Starter Snake for Replit

As discussed in the Feedback Repo (https://github.com/BattlesnakeOfficial/feedback/discussions/187) the current starter snake no longer runs directly on Replit. The error I get is below

```shell
> bash -c "[ -d ~/.cargo ] || bash <(curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs) -y --default-toolchain nightly; ~/.cargo/bin/cargo build; RUST_LOG=info target/debug/starter-snake-rust"
/nix/store/8kgsjv57icc18qhpmj588g9x1w34hi4j-bash-interactive-5.1-p12/bin/bash: line 1: /home/runner/.cargo/bin/cargo: No such file or directory
/nix/store/8kgsjv57icc18qhpmj588g9x1w34hi4j-bash-interactive-5.1-p12/bin/bash: line 1: target/debug/starter-snake-rust: No such file or directory
exit status 127
```

I believe this is happening because before Replit's Rust environment did not have a `~/.cargo` directory by default, so the run command used this to determine if rustup has been installed correctly.
It appears that Replit now has this directory, I tried switching the Run command to look for the full path to the cargo binary and this fixed the initial issue. However this requires installed the current Rust nightly in the run command, and my free Replit ran out of RAM before it could do that.

My next thought was that we might not need to install rustup if we didn't need to run on nightly rust. This was needed because Rocket, up until 0.5, has always required nightly Rust. So now that 0.5 is out in Release Candidate we should be able to move to stable rust, which this PR does!

Unfortunately this also didn't deploy to Replit, since the Rust version Replit came with was too old. Luckily Replit also supports packages from `nix` and the nix packages version of Rust and Cargo is more up to date! Up to date enough to run Rocket at least, which is all we needed.

I also added an app.json with a buildpack for Heroku [since a Procfile already existed]. I'm not sure if Heroku will actually auto-detect this correctly, but if not I think its good documentation! This is needed since Heroku doesn't have a Rust build-pack built in currently

### Try it out

Should be able to use this link to try out my fork on Replit! Warning it did take a while to download crates and compile for me but does run!

[https://replit.com/github/coreyja/starter-snake-rust](https://replit.com/github/coreyja/starter-snake-rust)

### CODE REVIEW

File: Procfile [hunk 1]
```diff
@@ -1 +1 @@
-web: ROCKET_PORT=$PORT ROCKET_KEEP_ALIVE=0 ./target/release/starter-snake-rust
```
Comment (coreyja) [inline Procfile:1]: These options are moved to application logic in `main.rs` or in the configuration file `Rocket.toml`

**Comment by coreyja** [comment]
Remembered this morning this needs some README changes! Will try to get those added here soon

File: .replit [hunk 2]
```diff
@@ -1,2 +1,2 @@
 language = "rust"
-run = "bash -c \"[ -d ~/.cargo ] || bash <(curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs) -y --default-toolchain nightly; ~/.cargo/bin/cargo build; RUST_LOG=info target/debug/starter-snake-rust\""
+run = "cargo run"
```
Comment (Larandar) [inline .replit:2]: By reverse engineering the default Rust repl this would be a complete `.replit` (with language server commented because poor 0.5vCPU)

```suggestion:-1
run = ["cargo", "run"]
entrypoint = "src/main.rs"

[packager]
language = "rust"

[packager.features]
packageSearch = true

[languages.rust]
pattern = "**/*.rs"


# From my personal experience rust-analyzer take everything on free tier
# [languages.rust.languageServer]
# start = "rust-analyzer"

```

**Review by Larandar — COMMENTED** [review]
Hey, so even if replit was too much of a friction for me to finally play here are a few changes that can help others :)

If you are updating the README, I would have like a warning about the fact that Rust work 10 000x better when compiled on developper machine and then deployed.

**Comment by bvanvugt** [comment]
@coreyja taking a look at this today, is there anything else you wanted to add?

**Comment by coreyja** [comment]
Oops @bvanvugt missed this ping earlier! I wanted to change something in the readme, got it changed now!

Didn't realize before it was only about a line but 🤷  Still better to get it updated lol
